main.go

```diff
@@ -3,147 +3,118 @@ package main
 import (
 	"bytes"
 	"context"
-	"encoding/json"
 	"fmt"
 	"log"
 	"net/http"
 	"net/url"
-	"os"
+	aws "oncall-notify/pkg/config/aws"
+	slack "oncall-notify/pkg/config/slack"
+	"oncall-notify/pkg/env"
 	"strings"
-	"time"
 
 	"github.com/aws/aws-lambda-go/lambda"
-	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/config"
-	"github.com/aws/aws-sdk-go-v2/service/ssmcontacts"
+	jsoniter "github.com/json-iterator/go"
 )
 
-const weekDuration = time.Hour * 24 * 7
+var config *Config
 
-var (
-	client           *ssmcontacts.Client
-	slackWebhookUrl  string
-	slackApiUrl      string
-	slackApiToken    string
-	slackSubTeamID   string
-	slackSubTeamName string
-	slackUserIDs     map[string]string
-)
+type Config struct {
+	AwsConfig   *aws.AwsConfig
+	SlackConfig *slack.SlackConfig
+}
 
-func getEnvOrDefault(key, defaultValue string) string {
-	value := os.Getenv(key)
-	if value == "" {
-		return defaultValue
+func NewConfig(awsConfig *aws.AwsConfig, slackConfig *slack.SlackConfig) *Config {
+	return &Config{
+		AwsConfig:   awsConfig,
+		SlackConfig: slackConfig,
 	}
-	return value
 }
 
-func init() {
-	awsRegion := getEnvOrDefault("AWS_REGION", "eu-west-1")
-	slackApiUrl = getEnvOrDefault("SLACK_API_URL", "https://slack.com/api/usergroups.users.update")
-	slackWebhookUrl = getEnvOrDefault("SLACK_WEBHOOK_URL", "")
-	slackApiToken = getEnvOrDefault("SLACK_API_TOKEN", "")
-	slackSubTeamID = getEnvOrDefault("SLACK_SUBTEAM_ID", "")
-	slackSubTeamName = getEnvOrDefault("SLACK_SUBTEAM_NAME", "support")
+func main() {
+	awsRegion := env.DefaultEnv("AWS_REGION", "eu-west-1")
+	slackApiUrl := env.DefaultEnv("SLACK_API_URL", "https://slack.com/api/usergroups.users.update")
+	slackWebhookUrl := env.DefaultEnv("SLACK_WEBHOOK_URL", "")
+	slackApiToken := env.DefaultEnv("SLACK_API_TOKEN", "")
+	slackSubTeamID := env.DefaultEnv("SLACK_SUBTEAM_ID", "")
+	slackSubTeamName := env.DefaultEnv("SLACK_SUBTEAM_NAME", "support")
+	slackUserIDs := slack.GetUserID()
 
 	if slackApiToken == "" {
-		log.Fatalf("error: Slack API token is not set")
+		fmt.Println("Error: Slack API token is not set")
+		return
```

Comment:
```suggestion
		log.Fatalf("error: Slack API token is not set")
```
Lambda should fail instead of continuing